### ISSUE TITLE
About the binarized weight

### ISSUE THREAD

**Issue by spoonnotfound** [jiecaoyu/XNOR-Net-PyTorch#63]
Hey, @jiecaoyu. I learnt a lot from your code. Thank you for your job. And I have two questions about the binarized weight.
First, I can see the binarized weight is stored in `target_module` list, but how can the model get and use these binarized weight for forward and backward? From this former [issue](https://github.com/jiecaoyu/XNOR-Net-PyTorch/issues/23#issuecomment-399336420), the model can get the binarized weight. I'd like to know why.
Second, I'm writing a loss function using XNOR-Net, and I need to get the binarized weight in the train phase. I cannot get the binarized weight by using `model.layer.weight.data`. How should I get it?

**Comment by jiecaoyu** [comment]
Hi @flyingpot 

1. You need to call the function [```bin_op.binarization()```](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/834cb6cf32e476f318a48ec73fe19d16058e2470/ImageNet/networks/main.py#L244) to get the binarized weights.

2. Please check these [lines](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/834cb6cf32e476f318a48ec73fe19d16058e2470/ImageNet/networks/util.py#L22)

**Comment by spoonnotfound** [comment]
Thank you @jiecaoyu. You solved my problems.
I got a new problem. Why does the model saved after `restore`? I think the saved model is not binarized, is it? And will this affect the result and make the accuracy higher?

**Comment by jiecaoyu** [comment]
It will not affect the results. Saving the model after ```restore``` is because we need to save the model for checkpoint. With only the binarized weights, we cannot resume the training.

**Comment by spoonnotfound** [comment]
Thank you! I got it.
